mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -53,9 +65,10 @@ class HashingTF(val numFeatures: Int) extends Serializable {
   @Since("1.1.0")
   def transform(document: Iterable[_]): Vector = {
     val termFrequencies = mutable.HashMap.empty[Int, Double]
+    val setTF = if (binary) (i: Int) => 1.0 else (i: Int) => termFrequencies.getOrElse(i, 0.0) + 1.0
```

Comment:
Could you please move this if-then-else outside of the loop?  (I'm guessing it would be worthwhile for speed.)